src/Tools/BuildValidator/Extensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace BuildValidator
+{
+    internal static class Extensions
+    {
+#if !NETCOREAPP
+        internal static void Write(this FileStream stream, ReadOnlySpan<byte> span)
+        {
+            for (int i = 0; i < span.Length; i++)
+            {
+                stream.WriteByte(span[i]);
```

Comment:
we are going to run on netframework as long as msbuild on windows invokes a netframework compiler. Is this really the best method at hand to write the span to the stream? For example, would changing the parameter from ReadOnlySpan to UnmanagedMemoryStream and calling `WriteTo(stream)` be a more reasonable cross-framework solution.

Reply:
Let me see if I can make the framework one better here. I'd like to keep the `Stream.Write(ReadOnySpan<byte>)` on .NET Core cause that is where the framework is going. BUt can make the framework downtarget solution bettere here.